raster-foundry/raster-foundry
# [Anchor] OGC services

## Overview

This PR anchors work to be done to add OGC support to backsplash.

It's opened with only WCS GetCapabilities for projects, treating `ProjectLayer`s as WCS layers.

### Checklist

Checklist items below are for the opening PR only. Separate PRs against this branch will have their own checklists.


### Demo

![wcs](https://user-images.githubusercontent.com/5702984/54386893-6bc70f80-4670-11e9-96c4-e17eaae6d448.gif)

### Notes

This won't build, since the gt-contrib and gt-server artifacts that it depends on aren't in the staging nexus. I'd like to put them there, but don't want to do this in an ad hoc way, and don't want to have staging merges depend on secret local only snapshots, so for now / until `feature/wms` is merged in geotrellis-server, I'll just demo things and make sure that artifacts are present for local development.

Still needs a second OGC service GetCapabilities endpoint, that's going to be WMS

## Testing Instructions

 * find a project with some layers (it's ok if some of them don't have imagery)
 * make it public
 * add `http://localhost:8081/wcs/<project id>` to WCS Layers in QGIS
 * you should get a list of your layer names
 * add `http://localhost:8081/wms/<project id>` to WMS/WMTS Layers in QGIS
 * you should get the same list

Closes #4745 

Closes #4747
